spec/integration/graphql_spec.rb

```diff
@@ -236,4 +236,217 @@
+        details: {
+          current: true,
+          started_on: Time.zone.local(2024, 7, 5),
+        },
+      ).tap do |role_appointment|
+        person = create(
+          :live_edition,
+          title: "The Rt Hon Sir Keir Starmer KCB KC MP",
+          document_type: "person",
+          schema_name: "person",
+          base_path: "/government/people/keir-starmer",
+          details: {
+            body: [{
+              content: "Sir Keir Starmer became Prime Minister on 5 July 2024.",
+              content_type: "text/govspeak",
+            }],
+          },
+        )
+        create(
+          :link_set,
+          content_id: role_appointment.content_id,
+          links_hash: { person: [person.content_id], role: [role.content_id] },
+        )
+      end
+
+      role_appointment2 = create(
+        :live_edition,
+        title: "The Rt Hon Rishi Sunak - Prime Minister",
+        document_type: "role_appointment",
+        schema_name: "role_appointment",
+        details: {
+          current: false,
+          started_on: Time.zone.local(2022, 10, 25),
+          ended_on: Time.zone.local(2024, 7, 5),
+        },
+      ).tap do |role_appointment|
+        person = create(
+          :live_edition,
+          title: "The Rt Hon Rishi Sunak MP",
+          document_type: "person",
+          schema_name: "person",
+          base_path: "/government/people/rishi-sunak",
+          details: {
+            body: [{
+              content: "Rishi Sunak was Prime Minister between 25 October 2022 and 5 July 2024.",
+              content_type: "text/govspeak",
+            }],
+          },
+        )
+        create(
+          :link_set,
+          content_id: role_appointment.content_id,
+          links_hash: { person: [person.content_id], role: [role.content_id] },
+        )
+      end
+
+      organisation1 = create(
+        :live_edition,
+        title: "Cabinet Office",
+        document_type: "organisation",
+        schema_name: "organisation",
+        base_path: "/government/organisations/cabinet-office",
+      )
+      organisation2 = create(
+        :live_edition,
+        title: "Prime Minister's Office, 10 Downing Street",
+        document_type: "organisation",
+        schema_name: "organisation",
+        base_path: "/government/organisations/prime-ministers-office-10-downing-street",
+      )
+      create(
+        :link_set,
+        content_id: role.content_id,
+        links_hash: {
+          ordered_parent_organisations: [organisation1.content_id, organisation2.content_id],
+          role_appointments: [role_appointment1.content_id, role_appointment2.content_id],
+        },
+      )
+    end
+
+    it "exposes Role-specific fields, generic Edition fields and Role's links" do
+      post "/graphql", params: {
+        query:
+          "{
+            edition(basePath: \"/government/ministers/prime-minister\") {
+              ... on Role {
+                basePath
```

Comment:
It turns out that we can easily make our fields snake-case... I think this would be nicer for our Rails-based consumers, but I don't know what the implications are for mobile apps.

Does anybody have an opinion about this?

Reply:
I mean i guess we're largely building this for ourselves, I'd say if it's easy enough we do it